lambda/custom/src/babybuddy/api.ts

```diff
@@ -26,14 +28,16 @@ const fetchSecrets: () => Promise<Secret> = async () => {
 
 class BabyBuddyApi {
   async getRequest<T>(url: string): Promise<GetResponse<T>> {
-    const { apiKey, apiUrl } = await fetchSecrets();
+    const { apiKey, apiUrl, cfAccessClientId, cfAccessClientSecret } = await fetchSecrets();
 
     try {
       const response = await axios.get(url, {
         baseURL: apiUrl,
         headers: {
           'Content-Type': 'application/json',
           Authorization: `Token ${apiKey}`,
+          'CF-Access-Client-Id': `${cfAccessClientId}`,
+          'CF-Access-Client-Secret': `${cfAccessClientSecret}`
         },
       });
```

Comment:
Lets create a separate base function for requests. In all three cases (`get`, `post`, and `delete`) the options passed to axios are the same:

```js
{
        baseURL: apiUrl,
        headers: {
          'Content-Type': 'application/json',
          Authorization: `Token ${apiKey}`,
          'CF-Access-Client-Id': `${cfAccessClientId}`,
          'CF-Access-Client-Secret': `${cfAccessClientSecret}`
        },
      }
```

So those options can established in the base function and the CF headers can be added conditionally only if set.